tests/nixos/rootless-daemon.nix

```diff
@@ -0,0 +1,177 @@
+{ pkgs, ... }:
+
+{
+  name = "rootless-daemon";
+
+  nodes.machine = { config, ... }: {
+    users.users.alice.isNormalUser = true;
+    #users.users.nix-daemon.isSystemUser = true;
+    users.users.nix-daemon.isNormalUser = true;
+    users.users.nix-daemon.group = "nix-daemon";
+    users.groups.nix-daemon = {};
+    environment.variables.NIX_REMOTE = "daemon"; # Even for root
+    virtualisation.writableStore = true;
+
+    boot.readOnlyNixStore = false;
+
+    # No root daemon
+    nix.enable = false;
```

Comment:
non-blocking:

Exposing a module for this kind of setup would be nice, but needs a few improvements.

I think you could blacklist the daemon module, at least on recent enough NixOS the daemon-specific stuff is separate from the Nix installation and config file.

I think we need a `nix.daemon.enable` in NixOS to be sure.